### PR TITLE
docs: update first step tutorial and templates info

### PR DESCRIPTION
**Pull request by lorenzo-cavazzi**

### CODE REVIEW

File: docs/user/templates.rst [hunk 1]
```diff
@@ -10,12 +10,13 @@ over. The two main aspects of the project that can be templated are:
...
-cases, you can use these templates as-is (see
-the :ref:`directory_structure` below). If the base template for either of
-these languages is not sufficient for you, or you require a different IDE,
-you can make modifications! Check out :ref:`advanced_interfaces`.
+When you create a project on the RenkuLab platform, you can choose between
+a few templates. You should see at least a Python setup (Basic Python Project)
+and a R setup (Basic R Project). In many cases, you can use these templates
+as-is (see the :ref:`directory_structure` below). If you prefer another
+language, you require a different IDE, or the base templates don't meet your
+needs for any other reason, you can make create new ones!
```
Comment (ciyer) [inline docs/user/templates.rst:18]: ` you can make create new ones` ->  `you can create new ones`

File: docs/user/templates.rst [hunk 2]
```diff
@@ -243,3 +244,58 @@ Note that some of these clash with the `renku` templates (i.e. content in
...
+* ``folder``: the target folder inside the repository where the template files
+  are stored. Please use a different folder for each template.
+* ``name``: a short user friendly name.
+* ``description``: a brief description of your template. This will be
+  presented to the user when choosing between templates.
+* ``variables``: we support the
+  `Jinja template engine <https://palletsprojects.com/p/jinja/>`_ in both
+  files content and filenames. You can therefore ask users for specific values
+  for any number of variables. The syntax is
+  ``<variable_name>: <vairable_description>``, where the name will be used as
```
Comment (ableuler) [inline docs/user/templates.rst:281]:
```suggestion
  ``<variable_name>: <variable_description>``, where the name will be used as
```

File: docs/user/templates.rst [hunk 3]
```diff
@@ -243,3 +244,58 @@ Note that some of these clash with the `renku` templates (i.e. content in
...
+  the variable name provided to the engine and the description will be
+  presented to the user to explain the variable's intended use.
+
+Use your repository
+"""""""""""""""""""
+
+If you installed the renku command line locally, you can provide your template
```
Comment (ableuler) [inline docs/user/templates.rst:288]:
```suggestion
If you installed the renku command line interface locally, you can provide your template
```

**Comment by lorenzo-cavazzi** [comment]
Thanks, changes applied

**Comment by ciyer** [comment]
Is this PR to adapt to the changes in the UI that come with https://github.com/SwissDataScienceCenter/renku-ui/pull/963?

File: docs/user/templates.rst [hunk 4]
```diff
@@ -243,3 +244,58 @@ Note that some of these clash with the `renku` templates (i.e. content in
...
+repository. Each of them requires an entry with the following parameters:
+
+* ``folder``: the target folder inside the repository where the template files
+  are stored. Please use a different folder for each template.
+* ``name``: a short user friendly name.
+* ``description``: a brief description of your template. This will be
+  presented to the user when choosing between templates.
+* ``variables``: we support the
+  `Jinja template engine <https://palletsprojects.com/p/jinja/>`_ in both
+  files content and filenames. You can therefore ask users for specific values
```
Comment (ciyer) [inline docs/user/templates.rst:279]: `files content` -> `file content`

**Comment by lorenzo-cavazzi** [comment]
@ciyer this is only the documentation. The (very few) changes needed in the renku repo are here #1259 .
Sorry for keeping all these PR separated, I hoped to keep the different topics separated but I see it's creatring confusion.

File: docs/user/templates.rst [hunk 5]
```diff
@@ -243,3 +244,58 @@ Note that some of these clash with the `renku` templates (i.e. content in
...
+  for any number of variables. The syntax is
+  ``<variable_name>: <variable_description>``, where the name will be used as
+  the variable name provided to the engine and the description will be
+  presented to the user to explain the variable's intended use.
+
+Use your repository
+"""""""""""""""""""
+
+If you installed the renku command line interface locally, you can provide
+your template repository to the ``renku init`` command. We recommend you to
```
Comment (ciyer) [inline docs/user/templates.rst:289]: `We recommend you to **always**...` -> `We recommend you **always**...`

**Comment by lorenzo-cavazzi** [comment]
@ciyer thanks for the review! I fixed the text with your suggestions.

**Pull request merged**